### PR TITLE
Add cause and trailers to GrpcError.

### PR DESCRIPTION
**Pull request by njovy**
Add a cause and trailers to GrpcError.

### CODE REVIEW

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* User @njovy isn't covered by a CLA. They will need to complete the form at https://identity.linuxfoundation.org/projects/cncf

_Regards,
CLA GitHub bot_

**Comment by njovy** [comment]
I signed it

File: lib/src/client/call.dart [hunk 1]
```diff
@@ -239,12 +240,15 @@ class ClientCall<Q, R> implements Response {
       // Only received a header frame and no data frames, so the header
       // should contain "trailers" as well (Trailers-Only).
       _trailers.complete(_headerMetadata);
-      final status = _headerMetadata['grpc-status'];
+      _headerMetadata.remove(":status");
+      final status = _headerMetadata.remove('grpc-status');
+//      final status = _headerMetadata['grpc-status'];
```
Comment (sigurdm) [inline lib/src/client/call.dart]: Remove out-commented code.

File: lib/src/client/call.dart [hunk 2]
```diff
@@ -239,12 +240,15 @@ class ClientCall<Q, R> implements Response {
       // Only received a header frame and no data frames, so the header
       // should contain "trailers" as well (Trailers-Only).
       _trailers.complete(_headerMetadata);
-      final status = _headerMetadata['grpc-status'];
+      _headerMetadata.remove(":status");
+      final status = _headerMetadata.remove('grpc-status');
```
Comment (sigurdm) [inline lib/src/client/call.dart]: Why do you remove the header here and above?

Comment (njovy) [reply lib/src/client/call.dart]: I tried to match what grpc-java returns when it calls a trailers in the same situation. I can just return a _headerMetadata.

**Comment by mit-mit** [comment]
Hi @njovy, thanks for the PR. Have you had a chance to update per the code review?

**Comment by njovy** [comment]
@mit-mit I wasn't sure if I need to keep or remove values from a metadata.

**Pull request closed without merging**